master/internal/experiment/postgres_experiments_intg_test.go

```diff
@@ -0,0 +1,167 @@
+//go:build integration
+// +build integration
+
+package experiment
+
+import (
+	"context"
+	"log"
+	"os"
+	"reflect"
+	"sort"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/require"
+
+	"github.com/determined-ai/determined/master/internal/db"
+	"github.com/determined-ai/determined/master/pkg/etc"
+)
+
+func TestMain(m *testing.M) {
+	pgDB, err := db.ResolveTestPostgres()
+	if err != nil {
+		log.Panicln(err)
+	}
+
+	err = db.MigrateTestPostgres(pgDB, "file://../../static/migrations", "up")
+	if err != nil {
+		log.Panicln(err)
+	}
+
+	err = etc.SetRootPath("../../static/srv")
+	if err != nil {
+		log.Panicln(err)
+	}
+
+	os.Exit(m.Run())
+}
+
+func TestPgDB_ExperimentCheckpointsToGCRawModelRegistry(t *testing.T) {
+	type args struct {
+		id             int
+		experimentBest int
+		trialBest      int
+		trialLatest    int
+	}
+
+	ctx := context.Background()
+
+	user := db.RequireMockUser(t, db.SingleDB())
+	exp := db.RequireMockExperiment(t, db.SingleDB(), user)
+	tr, task := db.RequireMockTrial(t, db.SingleDB(), exp)
+	a := db.RequireMockAllocation(t, db.SingleDB(), task.TaskID)
+	length := 4
+	var expectedCheckpoints []uuid.UUID
+	for i := 1; i <= length; i++ {
+		ckptUUID := uuid.New()
+		ckpt := db.MockModelCheckpoint(ckptUUID, a, db.WithSteps(i))
+		err := db.AddCheckpointMetadata(ctx, &ckpt, tr.ID)
+		require.NoError(t, err)
+		err = db.AddTrialValidationMetrics(ctx, ckptUUID, tr, int32(i), int32(i+5), db.SingleDB())
+		require.NoError(t, err)
+
+		if i == 2 { // add this checkpoint to the model registry
+			err = addCheckpointToModelRegistry(db.SingleDB(), ckptUUID, user)
+			require.NoError(t, err)
+		} else {
+			expectedCheckpoints = append(expectedCheckpoints, ckptUUID)
+		}
+	}
+
+	tests := []struct {
+		name    string
+		fields  db.PgDB
+		args    args
+		want    []uuid.UUID
+		wantErr bool
+	}{
+		{"test-000", *db.SingleDB(), args{exp.ID, 0, 0, 0}, expectedCheckpoints, false},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := ExperimentCheckpointsToGCRaw(ctx, tt.args.id, tt.args.experimentBest,
```

Comment:
style nit, take it or leave it. not really a fan of 'got'/'want' --> prob match the standard 'actual'/'expected'

Reply:
done